Sources/Common/Transform/LandmarkTransform/index.js

```diff
@@ -0,0 +1,328 @@
+        if(r != 0)
+        {
+          r = Math.sin(theta/2)/r;
+          x = x*r;
+          y = y*r;
+          z = z*r;
+        }
+        else // rotation by 180 degrees : special case
+        {
+          // Rotate around a vector perpendicular to ds
+          vtkMath.perpendiculars(ds, dt, 0, 0);
+          r = Math.sin(theta/2);
+          x = dt[0]*r;
+          y = dt[1]*r;
+          z = dt[2]*r;
+        }
+      }
+      else // points are not collinear
+      {
+        w = eigenVectors[0][0];
+        x = eigenVectors[1][0];
+        y = eigenVectors[2][0];
+        z = eigenVectors[3][0];
+      }
+
+      // convert quaternion to a rotation matrix
+
+      let ww = w*w;
+      let wx = w*x;
+      let wy = w*y;
+      let wz = w*z;
+
+      let xx = x*x;
+      let yy = y*y;
+      let zz = z*z;
+
+      let xy = x*y;
+      let xz = x*z;
+      let yz = y*z;
+
+      model.matrix[0] = ww + xx - yy - zz;
+      model.matrix[1] = 2.0*(wz + xy);
+      model.matrix[2] = 2.0*(-wy + xz);
+
+      model.matrix[4] = 2.0*(-wz + xy);
+      model.matrix[5] = ww - xx + yy - zz;
+      model.matrix[6] = 2.0*(wx + yz);
+
+      model.matrix[8] = 2.0*(wy + xz);
+      model.matrix[9] = 2.0*(-wx + yz);
+      model.matrix[10] = ww - xx - yy + zz;
+
+      if(mode != TransformMode.VTK_LANDMARK_RIGIDBODY)
+      {// add in the scale factor (if desired)
+        for(i = 0; i < 3; i++)
+        {
+          model.matrix[4*0 + i] *= scale;
+          model.matrix[4*1 + i] *= scale;
+          model.matrix[4*2 + i] *= scale;
+        }
+      }
+    }
+
+    // the translation is given by the difference in the transformed source
+    // centroid and the target centroid
+    let sx, sy, sz;
+
+    sx = model.matrix[0] * source_centroid[0] +
+         model.matrix[4] * source_centroid[1] +
+         model.matrix[8] * source_centroid[2];
+    sy = model.matrix[1] * source_centroid[0] +
+         model.matrix[5] * source_centroid[1] +
+         model.matrix[9] * source_centroid[2];
+    sz = model.matrix[2] * source_centroid[0] +
+         model.matrix[6] * source_centroid[1] +
+         model.matrix[10] * source_centroid[2];
+
+    model.matrix[12] = target_centroid[0] - sx;
+    model.matrix[13] = target_centroid[1] - sy;
+    model.matrix[14] = target_centroid[2] - sz;
+
+    // fill the bottom row of the 4x4 matrix
+    model.matrix[3] = 0.0;
+    model.matrix[7] = 0.0;
+    model.matrix[11] = 0.0;
+    model.matrix[15] = 1.0;
+  }
+
+}
+
+// ----------------------------------------------------------------------------
+// Object factory
+// ----------------------------------------------------------------------------
+
+const DEFAULT_VALUES = {
+  mode : Mode.SIMILARITY
+};
+
+// ----------------------------------------------------------------------------
+
+export function extend(publicAPI, model, initialValues = {}) {
+  Object.assign(model, DEFAULT_VALUES, initialValues);
+
+  // Internal objects initialization
```

Comment:
if vtkObject you should add `macro.obj(publicAPI, model)`.

If it's an vtkAlgorithm you should add the `macro.algo(publicAPI, model, numberOfInputs, numberOfOutputs)`

In which case, you should implement `requestData(in, out)` instead of update.